### ISSUE TITLE
Discussion: Promote attributes to entities?  Add State/Diagnostic/config entities

### ISSUE THREAD

**Issue by rct** [ufodone/envisalink_new#17]
There is a movement in Home Assistant for new integrations to not rely on attributes to store data that could arguably be entities in their own right.

(I can't find a good post to link to right now).

The old Envisalink integration creates a lot of attributes that could arguably be first class entities:

```
partition_state: N/A
alpha: Arm Away
ac_present: true
beep: false
bypass: false
chime: false
entry_delay: false
exit_delay: false
last_armed_by_user: 
last_disarmed_by_user: 
ready: false
bat_trouble: false
trouble: false
fire: false
alarm: false
alarm_fire_zone: false
alarm_in_memory: false
armed_away: true
armed_stay: false
armed_zero_entry_delay: false
icon: mdi:alarm
friendly_name: Home Keypad
```

I add template sensors to turn some of these into entities for (easier) use in automation. 

If this rewrite/update is going to result in breaking changes, maybe some of these attributes should be converted to primary state entities, diagnostic, or configuration entities?

For example:  Should `alarm`, `fire`, `ready`, `bypass`, `trouble`, `armed_away`, ... be their own binary sensors?   They seem like primary data.

**Comment by ufodone** [comment]
My original intention was that this upgrade would not introduce breaking changes.  To date, I **think** the only breaking change is with the service calls (`alarm_keypress` and `invoke_custom_function`).  (It might actually only be the latter.)

I'm reluctant to make more breaking changes given how much is already going into this update.  But I think it's certainly something to look at after the main changes are submitted.

It might also be something to poll other folks about.  I'm actually a pretty basic user of the integration so probably don't have the best perspective on what makes sense here.

**Comment by disforw** [comment]
I use FIRE, ALARM and TROUBLE as sensors. I also templated a switch for CHIME

**Comment by sheyman1** [comment]
Can you share how you switch CHIME on and off?  Been trying to figure out
how to do that.

Thanks

**Comment by disforw** [comment]
vista_panel_chime:
      friendly_name: "Vista Panel Chime Mode"
      value_template: "{{ state_attr('sensor.vista_panel_keypad', 'chime') }}"
      unique_id: 70000106
      icon_template: "mdi:bell"
      turn_on:
        service: envisalink.alarm_keypress
        data:
          entity_id: alarm_control_panel.vista_panel
          keypress: '41129'
      turn_off:
        service: envisalink.alarm_keypress
        data:
          entity_id: alarm_control_panel.vista_panel
          keypress: '41129'

## 4112 is the default installer code for a Vista panel, 9 is the button for chime

**Comment by hunstad** [comment]
re: It might also be something to poll other folks about

I expose the attribute ready via a template.  I am using a vista 21 and EVL 4 and I don't always get timely cleared/off notifications.

**Comment by sheyman1** [comment]
This looks great, but I have no idea what to do with it.  Can you please
tell me where to insert this code and how to use it?  Thank you, sorry I'm
still learning this...

Scott

**Comment by sheyman1** [comment]
Thank you very much for the info, very helpful.

**Comment by ufodone** [comment]
I have created a [beta](https://github.com/ufodone/envisalink_new/releases/tag/v0.8.0-beta1) release with a number of the attributes promoted to entities.  Feel free to try it out and let me know if there are any issues or suggestions.

**Comment by ufodone** [comment]
Released in [v0.8.0](https://github.com/ufodone/envisalink_new/releases/tag/v0.8.0)
